tests/sympc/tensor/mpc_tensor_test.py

```diff
@@ -472,3 +472,68 @@ def test_pow(get_clients, power) -> None:
 
     with pytest.raises(RuntimeError):
         power = x ** -2
+
+
+def test_backward(get_clients):
+    clients = get_clients(4)
+    session = Session(parties=clients)
+    session.autograd_active = True
+    SessionManager.setup_mpc(session)
+
+    x_secret = torch.tensor([[0.125, -1.25], [-4.25, 4], [-3, 3]], requires_grad=True)
+    y_secret = torch.tensor([[4.5, -2.5], [5, 2.25], [-3, 3]], requires_grad=True)
+    x = MPCTensor(secret=x_secret, session=session, requires_grad=True)
+    y = MPCTensor(secret=y_secret, session=session, requires_grad=True)
+
+    res_mpc = x * y
+    res = x_secret * y_secret
+    s_mpc = res_mpc.sum()
+    s = torch.sum(res)
+    s_mpc.backward()
+    s.backward()
+
+    assert np.allclose(x.grad.get(), x_secret.grad, rtol=1e-3)
+    assert np.allclose(y.grad.get(), y_secret.grad, rtol=1e-3)
+
+
+def test_backward_without_requires_grad(get_clients):
+    clients = get_clients(4)
+    session = Session(parties=clients)
+    session.autograd_active = True
+    SessionManager.setup_mpc(session)
+
+    x_secret = torch.tensor([[0.125, -1.25], [-4.25, 4], [-3, 3]])
+    y_secret = torch.tensor([[4.5, -2.5], [5, 2.25], [-3, 3]])
+    x = MPCTensor(secret=x_secret, session=session)
+    y = MPCTensor(secret=y_secret, session=session)
+
+    res_mpc = x - y
+    s_mpc = res_mpc.sum()
+    s_mpc.backward()
+
+    assert not res_mpc.requires_grad
+    assert x.grad is None
+    assert y.grad is None
+
+
+def test_backward_with_one_requires_grad(get_clients):
+    clients = get_clients(4)
+    session = Session(parties=clients)
+    session.autograd_active = True
+    SessionManager.setup_mpc(session)
+
+    x_secret = torch.tensor([[0.125, -1.25], [-4.25, 4], [-3, 3]], requires_grad=True)
+    y_secret = torch.tensor([[4.5, -2.5], [5, 2.25], [-3, 3]])
+    x = MPCTensor(secret=x_secret, session=session, requires_grad=True)
+    y = MPCTensor(secret=y_secret, session=session)
+
+    res_mpc = x - y
+    res = x_secret - y_secret
+    s_mpc = res_mpc.sum()
+    s = torch.sum(res)
+    s_mpc.backward()
+    s.backward()
+
+    assert res_mpc.requires_grad
+    assert np.allclose(x.grad.get(), x_secret.grad, rtol=1e-3)
```

Comment:
Q: Why not add the check for also `res.grad` and `res_mpc.grad`?

Reply:
added an issue for this